src/simularium/VisGeometry.ts

```diff
@@ -1259,24 +961,62 @@ class VisGeometry {
     }
 
     private setGeometryData(typeMapping: EncodedTypeMapping): void {
-        this.logger.debug("Received type mapping data: ", typeMapping);
+        this.logger.info("Received type mapping data: ", typeMapping);
         Object.keys(typeMapping).forEach((id) => {
             const entry: AgentDisplayDataWithGeometry = typeMapping[id];
-            this.mapIdToGeom(
-                Number(id),
-                entry.geometry.displayType,
-                entry.geometry.url,
-                entry.geometry.color
-            );
+            const { url, displayType } = entry.geometry;
+            const lookupKey = url ? checkAndSanitizePath(url) : displayType;
+            this.visGeomMap.set(Number(id), lookupKey);
+            this.geometryStore
+                .mapKeyToGeom(Number(id), entry.geometry)
+                .then((returned) => {
+                    if (!returned) {
+                        // no new geometry to load
```

Comment:
when does this happen?

Reply:
If the same geometry is assigned to the two different agent ids, so there is no new geometry needed to load.

Reply:
i see, so returned could be thought of as something like "newGeometry", roughly

Reply:
yeah, I added some comments and renamed it so hopefully this will be clearer